cpp/src/groupby/groupby.cu

```diff
@@ -55,6 +55,9 @@ groupby::groupby(table_view const& keys,
     _column_order{column_order},
     _null_precedence{null_precedence}
 {
+  auto is_list  = [](auto const& col) { return col.type().id() == cudf::type_id::LIST; };
+  auto has_list = std::any_of(keys.begin(), keys.end(), is_list);
+  CUDF_EXPECTS(not has_list, "Groupby does not support LIST columns as keys.");
```

Comment:
This check should be put in a more appropriate place. Perhaps the constructor of the `row_equality_comparator` which should verify that all of the columns passed in are type that is equality comparable.

Reply:
Oh, yes. Far more elegant.

Reply:
Hmm. A minor wrinkle. Here's the signature for `row_equality_comparator`'s constructor:
```c++
  row_equality_comparator(table_device_view lhs, table_device_view rhs, bool nulls_are_equal = true)
    : lhs{lhs}, rhs{rhs}, nulls_are_equal{nulls_are_equal}
  {...}
```
This works on `table_device_view`, which does not allow access/iteration on the columns from `__host__`. :/